desktop_version/src/UtilityClass.cpp

```diff
@@ -225,13 +225,20 @@ void UtilityClass::updateglow()
 
 bool is_number(const char* str)
 {
-	for (int i = 0; str[i] != '\0'; i++)
+	if (str[0] == '\0')
 	{
-		if (!SDL_isdigit(static_cast<unsigned char>(str[i])) && (i != 0 || str[0] != '-'))
+		return false;
+	}
+
+	for (size_t i = 0; str[i] != '\0'; ++i)
+	{
+		if (!SDL_isdigit(str[i])
+		&& (i != 0 || str[0] != '-' || str[1] == '\0'))
```

Comment:
Do we just want to unroll iterations 0 and 1? It took me a few tries to figure out even the original loop check...

Reply:
Yeah, we can unroll it. Just give me a second...

Reply:
Took me a bit, but I think I've figured it out. Rebased and force-pushed. (Also fixed a bug with `VVV_isxdigit()`.)